src/utils.ts

```diff
@@ -10,10 +10,13 @@ export function isRelative(inputString: string) {
 const PROTOCOL_STRICT_REGEX = /^\w{2,}:([/\\]{1,2})/;
 const PROTOCOL_REGEX = /^\w{2,}:([/\\]{2})?/;
 const PROTOCOL_RELATIVE_REGEX = /^([/\\]\s*){2,}[^/\\]/;
+const PROTOCOL_SCRIPT_RE = /^(data|javascript|vbscript):$/;
```

Comment:
I am not sure if `data` is even a valid protocol in browsers that we need to disable. (Because consequence is we we want to enable safe feature by default at some point, it breaks valid URLs used for ESM imports and image URLs, etc)

Reply:
Check out: https://github.com/github/codeql/blob/8438fb231098a9c142ccf9f9019b73c1df827b89/javascript/ql/src/Security/CWE-020/IncompleteUrlSchemeCheck.qhelp#L6-L18.

I'm not sure I would recommend enabling by default within `hasProtocol`.

Reply:
I am wondering why not reverse protocol check via an "allowlist" then? (only allow `http(s)` for external redirexcts). I am not sure if putting validation logic inside hasProtocol is best idea since `data:` is a valid protocol only not **secure** to blindly trust for an external redirect.

Reply:
Reversing the check doesn't quite work, because people might want to redirect to random (and unforeseeable) protocols like `web+ap:` legitimately.

`javascript:` is also a valid protocol, just like `data:`. The issue is that these three protocols have the possibility of executing code when loaded.

Reply:
My main worry is - what do we mean by a 'safe' protocol?

Reply:
It can be `isJSProtocol` or something then.. I mainly prefer to separate these two utils from each other. Regardless what we call it or how much tighten it. Having protocol and validating it are two different usages.

Reply:
That makes sense 👍

Reply:
`blob:` is another potentially dangerous protocol, could in theory trigger XSS